.swm/frontend-structure.c1t7o.sw.md

```diff
@@ -0,0 +1,89 @@
+---
+id: c1t7o
+name: Frontend Structure
+file_version: 1.0.2
+app_version: 0.8.5-0
+file_blobs:
+  static/app/sentryTypes.tsx: 0fe070ac5d22d1d33a58de460c6c56b3b9841abf
+---
+
+The frontend codebase is under `📄 static/app` . Our framework is React. There are additional folders related to the frontend in additional locations, as described below.
+
+The main folders to know are:
+
+# Components
+
+We use components for UI that are designed to be highly reusable.
+
+## `📄 static/app/components`
+
+Components are located under this folder - for example `📄 static/app/components/idBadge` .
+
+Placing an `index` file in a component folder provides a way to implicitly import the main file without specifying it.
+
+If the folder is created to group components that are used together, and there is an entrypoint component, that uses the components within the grouping - the entrypoint component should be the index file. For example, see `📄 static/app/components/idBadge/index.tsx`
+
+## `📄 docs-ui/stories/components`
+
+Note that every component should have a corresponding `.stories.js` file that documents how it should be used. For example, `📄 static/app/components/idBadge` has the corresponding file `📄 docs-ui/stories/components/idBadge.stories.js`
+
+## `📄 tests/js/spec/components`
+
+Tests for components. E.g., `📄 tests/js/spec/components/idBadge`
+
+# Views
+
+We use views for UI that will typically not be reused in other parts of the codebase.
+
+## `📄 static/app/views`
+
+Views are located under this folder.
+
+## `📄 tests/js/spec/views`
+
+Tests for views. E.g., `📄 tests/js/spec/views/accountClose.spec.jsx` for `📄 static/app/views/settings/account/accountClose.tsx` .
+
+# State Management
+
+We use [Reflux](https://github.com/reflux/refluxjs) for managing our global state. Reflux implements the unidirectional data flow pattern outlined by [Flux](https://facebook.github.io/flux/).
+
+*   `📄 static/app/stores` - stores are registered here, and are used to store various pieces of data used by the application. For example, `📄 static/app/stores/groupStore.tsx`
+    
+*   `📄 static/app/actions` - actions are registered here, e.g., `📄 static/app/actions/groupActions.tsx` .
+    
+*   `📄 static/app/actionCreators` - action creator functions are used to dispatch actions. For example, `📄 static/app/actionCreators/group.tsx`. Reflux stores listen to actions and update themselves accordingly.
+    
+
+To learn more about our use of Reflux, refer to [State Management with Reflux](state-management-with-reflux.jni2e.sw.md) .
+
+# Important Files
+
+## `📄 static/app/main.tsx`
+
+This is the web app, it is intended to be available online.
```

Comment:
How is one file 'the web app'? Maybe 'the web app entrypoint'? also the 'intented to be available online sentence is unclear to me